### ISSUE TITLE
suggestion to add a potential memory leak warning to the readme

### ISSUE THREAD

**Issue by moshetet** [Coggle/s3-streamlogger#29]
S3StreamLogger will keep the logs in this.buffers until it needs to create a new file.
So if setting large rotate_every and max_file_size you can bump into a nasty memory leak.

A warning should be added to the README to warn against this, especially since the following 2 things cause you to believe it is not an option unless you dig into the code of the package:
1. The package says that it implements a stream (which usually means things don't stay in memory) after they are sent away. 
2. The  options rotate_every and max_file_size just refer to the way the files are rotated in S3.

**Comment by autopulated** [comment]
In order to update an existing file in S3, this module **re-uploads the entire file**. Due to the way S3 works, it is not practical to do anything else. For this reason you should use a relatively small maximum file size.

**Comment by conor-deegan** [comment]
Once a file rotates or reaches it's max_file_size or rotates in S3, are the logs cleared from this.buffer?

Once a file rotates or reaches it's maximum do the logs on the application clear?

**Comment by autopulated** [comment]
Yes, buffers are cleared when a file is rotated due to reaching maximim age or size.

**Comment by conor-deegan** [comment]
Thank you!

**Comment by vks-dbb** [comment]
is there might be some potential log missing in parallel executing?

**Comment by autopulated** [comment]
@vks-dbb Yes, different servers/processes should write log files to different files. The default `name_format` tries to ensure this.
